kpatch-build/kpatch-build

```diff
@@ -385,7 +385,7 @@ find_parent_obj() {
 	PARENT="$(basename "$parent")"
 	PARENT="${PARENT#.}"
 	PARENT="${PARENT%.cmd}"
-	PARENT="$dir/$PARENT"
+	[[ $dir != "." ]] && PARENT="$dir/$PARENT"
```

Comment:
Why don't we want './' here? It shouldn't hurt so I don't see any reason for this change.

Reply:
> Why don't we want './' here? It shouldn't hurt so I don't see any reason for this change.

For the "dir" is get from dir="$(dirname "$parent")", it would be either "." or "xxxx/yyy/".

Reply:
But why can't we just leave `PARENT="$dir/$PARENT"` so that if $dir is "." we get `PARENT="./$PARENT"`? This is a completely valid path so we can have a unified solution for both cases without doing this check.

Reply:
> But why can't we just leave `PARENT="$dir/$PARENT"` so that if $dir is "." we get `PARENT="./$PARENT"`? This is a completely valid path so we can have a unified solution for both cases without doing this check.

The original input parameter of "find_parent_obj" is got from the file "changed_objs", which maybe either "x1.o" or "dir1/x2.o".
for the "x1.o",  the result of the cmd "dirname x1.o" is "."，and than we would get the parant with "./", such as "./y1.o", therefore when we add "-w" for grep, the result of the cmd  "grep -lw "./y1.o" ./.*.cmd" is none. This is why we don't want "./".
for the "dir1/x2.o", the result of the cmd "dirname dir1/x2.o" is "dir1", and than we would get
the parant with "dir1/", such as "dir1/y2.o".

Reply:
Oh, I see now. Thank you for the explanation.